### PR TITLE
Added type definition for passport-http

### PR DESCRIPTION
**Pull request by krizalys**
Added a new type definition + test for [passport-http](https://github.com/jaredhanson/passport-http):
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

### CODE REVIEW

**Comment by vvakame** [comment]
*passport-http/passport-http.d.ts*

Checklist

* [X] is correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)?
  * https://www.npmjs.com/package/passport-http - https://github.com/jaredhanson/passport-http
* [X] has a [test file](http://definitelytyped.org/guides/contributing.html#tests)? (passport-http-tests.ts or passport-http-tests.tsx)
* [ ] pass the Travis CI test?

**Comment by vvakame** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request
Now, we are working on `types-2.0` branch.
Could you send this change to that branch?

**Comment by krizalys** [comment]
Rebased.

**Comment by krizalys** [comment]
@vvakame

**Comment by vvakame** [comment]
could you rename `passport-http/passport-http.d.ts` to `passport-http/index.d.ts`?

**Comment by vvakame** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package
add tsconfig.json.

File: passport-http/passport-http.d.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+/// <reference path="../passport/passport.d.ts"/>
+/// <reference path="../express/express.d.ts"/>
+
+declare module "passport-http" {
+
+    import passport = require("passport");
+    import express = require("express");
+
+    interface IBasicStrategyOptions {
```
Comment (vvakame) [inline passport-http/passport-http.d.ts]: could you remove `I` prefix?

**Comment by krizalys** [comment]
Thanks for the comments @vvakame, I missed these recommendations at first glance.

I proceeded to the general revision requested, hope you'll like it better.

**Pull request merged**